storage/flash_common.c

```diff
@@ -115,13 +117,57 @@ secbool flash_area_write_word(const flash_area_t *area, uint32_t offset,
   return flash_write_word(sector, sector_offset, data);
 }
 
-secbool flash_area_write_quadword(const flash_area_t *area, uint32_t offset,
-                                  const uint32_t *data) {
+secbool flash_area_write_block(const flash_area_t *area, uint32_t offset,
+                               const flash_block_t block) {
+  if (!FLASH_IS_ALIGNED(offset)) {
+    return secfalse;
+  }
+
+  uint16_t sector;
+  uint32_t sector_offset;
+  if (sectrue != get_sector_and_offset(area, offset, &sector, &sector_offset)) {
+    return secfalse;
+  }
+
+#if FLASH_BLOCK_WORDS == 1
+  return flash_write_word(sector, sector_offset, block);
+#else
+
+  uint32_t *addr =
+      (uint32_t *)flash_area_get_address(area, offset, 4 * sizeof(uint32_t));
+
+  if (addr == NULL) {
+    return secfalse;
+  }
+
+  secbool old_all_ff = sectrue;
+  secbool new_all_00 = sectrue;
+  secbool all_equal = sectrue;
+
   for (int i = 0; i < 4; i++) {
```

Comment:
4 should be `FLASH_BLOCK_WORDS`

Reply:
fixed https://github.com/trezor/trezor-firmware/commit/6e1e42d6e01cc267862ef3571581a548030d7dab